lib/set_member_subject_selector.rb

```diff
@@ -11,7 +11,7 @@ def initialize(workflow, user)
 
   def set_member_subjects
     to_classify = select_set_member_subjects_to_classify
-    unless to_classify.exists?
+    if !to_classify.exists? && user
```

Comment:
Do you want to do this check the other way around? `user && !to_classify.exists?` it should save a db query to check to_classify in cases where no user is supplied.